src/android/unregisterNativeModule.js

```diff
@@ -1,75 +1,52 @@
-const readFile = require('./fs').readFile;
-const writeFile = require('./fs').writeFile;
-const path = require('path');
-const compose = require('lodash').flowRight;
+const fs = require('fs');
 const getReactVersion = require('../getReactNativeVersion');
 const getPrefix = require('./getPrefix');
 const isInstalled = require('./isInstalled');
 
-const cut = (scope, pattern) =>
-  scope.replace(pattern, '');
+const revokePatch = require('./patches/revokePatch');
+const makeSettingsPatch = require('./patches/makeSettingsPatch');
+const makeBuildPatch = require('./patches/makeBuildPatch');
+const makeStringsPatch = require('./patches/makeStringsPatch');
 
-module.exports = function unregisterNativeAndroidModule(name, dependencyConfig, projectConfig) {
-  const prefix = getPrefix(getReactVersion(projectConfig.folder));
-
-  /**
-   * @param  {String} content Content of the Settings.gradle file
-   * @return {String}         Patched content of Settings.gradle
-   */
-  const cutModuleFromSettings = (name) => (content) =>
-    cut(content, `include ':${name}'\n` +
-      `project(':${name}').projectDir = ` +
-      `new File(rootProject.projectDir, '../node_modules/${name}/android')\n`
-    );
-
-  /**
-   * Cut module compilation from the project build
-   * @param  {String} content Content of the Build.gradle file
-   * @return {String}         Patched content of Build.gradle
-   */
-  const cutModuleFromBuild = (name) => (content) =>
-    cut(content, `    compile project(':${name}')\n`);
-
-  const getAddPackagePatch = require(`./${prefix}/addPackagePatch`);
+module.exports = function unregisterNativeAndroidModule(
+  name,
+  androidConfig,
+  projectConfig
+) {
 
-  /**
-   * Make a MainActivity.java program patcher
-   * @param  {String}   importPath Import path, e.g. com.oblador.vectoricons.VectorIconsPackage;
-   * @param  {String}   instance   Code to instance a package, e.g. new VectorIconsPackage();
-   * @return {Function}            Patcher function
-   */
-  const makeMainActivityPatcher = (content) => {
-    const patched = cut(content, dependencyConfig.packageImportPath + '\n');
-    return cut(patched, getAddPackagePatch(dependencyConfig));
-  };
+  if (!isInstalled(projectConfig, name)) {
```

Comment:
That was a follow up to this comment https://github.com/rnpm/rnpm-plugin-link/pull/70#discussion_r60419334. Basically `unlink` is not yet refactored and this check is OK. I haven't made it yet mainly because I was waiting for your branch (this branch)